### ISSUE TITLE
Clarifications to Readme.md and Upgrading.md

### ISSUE THREAD

**Issue by bill-orange** [m0ngr31/kodi-alexa#171]
Being a novice, there are a few things I noticed in the instructions that might not register as confusing to a more experienced used.

Readme.md:

"If you are self-hosting, you will need to create the file kodi.config from the kodi.config.example template. The template file contains comments to describe the options."

I believe that this step is necessary regardless of whether or not you are "self hosting".  Moreover, shouldn't there be a copy of the template in the code files rather than just the link in the readme?

I think that regarding slot building; "Note that this will communicate with the Kodi instance defined in the [DEFAULT] section in kodi.config." Should also say: Therefore, you must have completed and saved at least the  [DEFAULT] section of kodi.config before using generate_custom_slots.py.  Seems obvious but it tripped me up.

Upgrading.md:

I would note somewhere that .env is no long used and kodi.config must be created if the user has not already done so.   I know that there is the general mention that anything my have changed, but this one is really important and deserves a special mention.

Thanks. This skill is a great piece of work.  I have found reviewing the code is very helpful in learning how to write skills.

**Comment by jingai** [comment]
> "If you are self-hosting, you will need to create the file kodi.config from the kodi.config.example template. The template file contains comments to describe the options."
>
> I believe that this step is necessary regardless of whether or not you are "self hosting".

No, it is only necessary for AWS Lambda and self-hosting.  There's a copy of the statement in the Lambda section.

The section of the documentation you're referring to assumes the user has already gone through the Lambda section if they're hosting with AWS.

> Moreover, shouldn't there be a copy of the template in the code files rather than just the link in the readme?

The main copy of this file now resides in a library that we link kodi-alexa to.  Hence, we provide a link in the README to the file (which is outside this repo).

> I think that regarding slot building; "Note that this will communicate with the Kodi instance defined in the [DEFAULT] section in kodi.config." Should also say: Therefore, you must have completed and saved at least the [DEFAULT] section of kodi.config before using generate_custom_slots.py. Seems obvious but it tripped me up.

Yes, we assume you actually completed the previous step(s).  If you deployed without saving kodi.config, you've already messed up.

In a nutshell:  follow the steps in the README _in order please_.  If you skip around, we can't acount for every thing you might have missed.

> Upgrading.md:
>
> I would note somewhere that .env is no long used and kodi.config must be created if the user has not already done so. I know that there is the general mention that anything my have changed, but this one is really important and deserves a special mention.

Aye, we could be more specific with upgrading instructions.

**Comment by quadhammer** [comment]
These clarifications might help the readme:

Where it says:
"On the next page, you'll have to paste the contents of the IntentSchema.json file into the "Intent Schema" field, and paste the contents of the SampleUtterances.txt file in the "Sample Utterances" field. " These files are located in the "https://github.com/m0ngr31/kodi-alexa/tree/master/speech_assets" directory.

**Comment by jingai** [comment]
@quadhammer, I'll add this in for the next release.

**Comment by jingai** [comment]
Links to the intent and utterances files added in edd08bd0f48e6219e99ce4bb521ebd0aaeb140e2

**Comment by jingai** [comment]
I think the rest of this boils down to "our upgrade procedures could probably use some TLC."

**Comment by jingai** [comment]
Besides the note about .env -> kodi.config (which is a one-time thing, so I'm inclined to just ignore this one for now), what else do you all think should be added to UPGRADING?

**Comment by jingai** [comment]
If there's something specific you think should be added to UPGRADING to make it more clear, please open a new Issue describing what you're having trouble with.
